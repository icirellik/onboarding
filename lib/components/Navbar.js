import React from 'react';
import PropTypes from 'prop-types';
import { Navbar, Nav, NavItem } from 'react-bootstrap';

const styles =  {
  navBarHeader: {
    color: "#fff",
    fontSize: "3em",
  },
  navBar: {
    backgroundColor: "#44C938",
  },
};

class AppNavbar extends React.PureComponent {
  render() {
    return (
      <Navbar style={styles.navBar}>
        <Navbar.Header style={styles.navBarHeader}>
          {this.props.i18n.header}
        </Navbar.Header>
      </Navbar>
    );
  }
}

AppNavbar.propTypes = {
  i18n: PropTypes.object.isRequired,
}

export default AppNavbar;
