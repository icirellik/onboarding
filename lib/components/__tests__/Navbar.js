import React from 'react';
import Navbar from '../Navbar';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Navbar', () => {

  const testProps = {
    i18n: {
      header: "Header"
    },
  };

  it('renders correctly', () => {

    const wrapper = shallow(
      <Navbar
        {...testProps}
      />
    );

    expect(wrapper.find('NavbarHeader').length).toBe(1);
    expect(wrapper).toMatchSnapshot();

  });

});
