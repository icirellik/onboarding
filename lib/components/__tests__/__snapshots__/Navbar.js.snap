// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppNavbar
    i18n={
      Object {
        "header": "Header",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsStyle": "default",
      "children": <NavbarHeader
        style={
          Object {
            "color": "#fff",
            "fontSize": "3em",
          }
        }
      >
        Header
      </NavbarHeader>,
      "style": Object {
        "backgroundColor": "#44C938",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Header",
        "style": Object {
          "color": "#fff",
          "fontSize": "3em",
        },
      },
      "ref": null,
      "rendered": "Header",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsStyle": "default",
        "children": <NavbarHeader
          style={
            Object {
              "color": "#fff",
              "fontSize": "3em",
            }
          }
        >
          Header
        </NavbarHeader>,
        "style": Object {
          "backgroundColor": "#44C938",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Header",
          "style": Object {
            "color": "#fff",
            "fontSize": "3em",
          },
        },
        "ref": null,
        "rendered": "Header",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
